### PR TITLE
fix: throttle event listener function

### LINKED ISSUE #686: [Bug] multiple header-labels are added when creating a new issue without a template
### Version

1.9.0

### Current Behavior

When a new issue is created from the created issue page and the repository does not have a configuration template, multiple header-label are added to the form page.

### Expected Behavior

only one header-label are added to the form page.

### Environment

```markdown
- OS: Win11
- Browser: Edge
```

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by wxharry**
## Brief Information

This pull request is in the type of ([more info](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#type) about types):

- [ ] build
- [ ] ci
- [ ] docs
- [ ] feat
- [x] fix
- [ ] perf
- [ ] refactor
- [ ] test

Related issues ([all available keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

- fix #686  

## Details

1. add a debounce function 
2. debouce the function added to the `turbo:render` event
## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/hypertrons/hypertrons-crx/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/hypertrons/hypertrons-crx)

### CODE REVIEW

File: src/feature-manager.ts [hunk 1]
```diff
@@ -176,33 +176,48 @@ const add = async (
...
+     * response to an event. It ensures that the action is executed only once after a certain
+     * duration of inactivity, even if the event is fired multiple times within that duration.
+     *  */
+    const debounce = function (func: Function, delay: number) {
+      let debounceTimer: any;
+      return function () {
+        clearTimeout(debounceTimer);
+        debounceTimer = setTimeout(func, delay);
+      };
+    };
```
Comment (tyn1998) [inline src/feature-manager.ts:190]: Hi @wxharry, the [debounce helper](https://github.com/hypertrons/hypertrons-crx/blob/94bcf7231b7c82db6f477f7859c87f5d8b5c3a5f/src/helpers/debounce.ts) is already in the code base :D

Comment (wxharry) [reply src/feature-manager.ts:190]: Oh, I didn't notice that. Will replace with the existing code

**Comment by andyhuang18** [comment]
I can't view the chart of `Starred`.

https://github.com/hypertrons/hypertrons-crx/assets/50283262/aaf3feac-7058-4c1d-bc51-5fe45083eb0a

**Comment by lhbvvvvv** [comment]
There are no bugs on mine. I can view the chart of `Starred`.
xx

**Comment by andyhuang18** [comment]
> I can't view the chart of `Starred`.
> 
>  2023-07-17.12-44-01.mp4

I made a mistake, so far everything is OK!

**Comment by tyn1998** [comment]
Hey @wxharry, thanks for your solution!

This solution works, however, it delays all loadings of features after tab switches. If `500` is set to `5000`, the delay can be noticed very well. I guess a simple `await sleep(500)` will work too.

Instead of `debounce` I think `throttle` can be used here to solve the issue. The latter doesn't wait for the specified time when the throttled function is triggered at first time.

**Comment by wxharry** [comment]
> Hey @wxharry, thanks for your solution!
> 
> This solution works, however, it delays all loadings of features after tab switches. If `500` is set to `5000`, the delay can be noticed very well. I guess a simple `await sleep(500)` will work too.
> 
> Instead of `debounce` I think `throttle` can be used here to solve the issue. The latter doesn't wait for the specified time when the throttled function is triggered at the first time.

Hey @wxharry , I have implemented the throttle function for the feature adding. Would you mind testing again?

File: src/helpers/throttle.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  const throttledFunc = (args: A): Promise<R> =>
+    new Promise((resolve) => {
+      if (!isThrottled) {
+        isThrottled = true;
+        resolve(fn(args));
+
+        setTimeout(() => {
+          isThrottled = false;
+        }, ms);
```
Comment (tyn1998) [inline src/helpers/throttle.ts:15]: Hi @wxharry, I think this implementation of throttle has a bug: if the `throttledFunc` is continuously called every `ms-0.000...1` ms, then the `fn` will never be called. I mean, once the `isThrottled` is set to `true`, succeeding calls of `throttledFunc` should not refresh the `isThrottled`.

Don't worry, I suppose these basic functions may not be written by our own. I'll push a commit to substitue them with [lodash's](https://lodash.com/docs/4.17.15#throttle).

**Review by tyn1998 — CHANGES_REQUESTED** [review]

**Pull request merged**